StRoot/StETofUtil/StETofGeometry.cxx

```diff
@@ -1488,4 +1572,82 @@ StETofGeometry::getFieldZ( const double& x, const double& y, const double& z ) {
     mStarBField->BField( X, B );
 
     return B[ 2 ];
-}
\ No newline at end of file
+}
+
+void 
+StETofGeometry::readAlignmentParameters(){
+      LOG_INFO << "etofAlignParam: filename provided --> use parameter file: " << mFileNameAlignParam.c_str() << endm;
+        
+		  //add check for no alignment set here.
+	
+		  ifstream paramFile;		
+        paramFile.open( mFileNameAlignParam.c_str() );
+
+        if( !paramFile.is_open() ) {
+            LOG_INFO << "unable to get the alignments parameters from file --> file does not exist" << endm;
+            return;
+        }
+
+	     StThreeVectorD counterAlignmentParameter;
+        float tempX;
+		  float tempY;
+		  float tempZ;
+        while( paramFile >> tempX >> tempY >> tempZ ) {
+		  counterAlignmentParameter = StThreeVectorD( tempX, tempY, tempZ );
+            mAlignmentParameters.push_back( counterAlignmentParameter );
+        }
+        
+        paramFile.close();
+
+        if( mAlignmentParameters.size() != 108 ) {
+            LOG_INFO << "parameter file for alignments has not the right amount of entries: ";
+            LOG_INFO << mAlignmentParameters.size() << " instead of 108 !!!!" << endm;
+            return;
+        }
+
+}
+
+void 
+StETofGeometry::readAlignmentDatabase(StMaker* maker){
+        LOG_INFO << "No etof alignment file provided --> use database: " << endm;
+        
+		  //add check for no alignment set here.
+	
+    	  TDataSet* dbDataSet = nullptr;
+
+		 if(!maker){
+			LOG_WARN << "No valid pointer to any StMaker given. Do not load alignment from database" << endm;
+			return;
+		 }
+
+        dbDataSet = maker->GetDataBase( "Geometry / etof / etofAlign" );
```

Comment:
I don't think you need a instance of a maker to access a data set. See if there is a static method

Reply:
Fixed. Thanks for finding the proper way of doing it